### PR TITLE
Add anchor events for developer centric timeline

### PR DESCRIPTION
**Pull request by B3rn475**
By adding these events the new developer centric timeline will be able
to identify and correlate events related to the different stages of the
drawing pipeline.

This Pull Request is for review purposes, it needs to wait for the corresponding changes in the Dart VM to land first.

### CODE REVIEW

File: shell/platform/android/vsync_waiter_android.cc [hunk 1]
```diff
@@ -59,8 +59,17 @@ static void OnNativeVsync(JNIEnv* env,
...
-  TRACE_EVENT0("flutter", "VSYNC");
-  TRACE_EVENT_INSTANT0("flutter", "VSYNC");
+  {
+      // Every 10 bits we can potentially increase the size of the output
+      // by 3 digis.
+      // sizeof(jlong) * 8 (number of bits)
+      // sizeof(jlong) * 8 / 10 (number of groups of 10 bits)
+      // sizeof(jlong) * 8 / 10 * 3 (number of digits - 1)
+      // sizeof(jlong) * 8 / 10 * 3 + 2 (number of digits + \0)
+      char deadline[sizeof(jlong) * 8 / 10 * 3 + 2];
```
Comment (a-siva) [inline shell/platform/android/vsync_waiter_android.cc]: Can these two lines also be made conditional under profile/debug

File: shell/platform/darwin/ios/framework/Source/vsync_waiter_ios.mm [hunk 2]
```diff
@@ -54,6 +54,22 @@ - (void)onDisplayLink:(CADisplayLink*)link {
...
+  // See: https://github.com/catapult-project/catapult/blob/2091404475cbba9b786
+  // 442979b6ec631305275a6/tracing/tracing/extras/vsync/vsync_auditor.html#L26
+  {
+    // Every 10 bits we can potentially increase the size of the output
+    // by 3 digis.
+    // sizeof(jlong) * 8 (number of bits)
+    // sizeof(jlong) * 8 / 10 (number of groups of 10 bits)
+    // sizeof(jlong) * 8 / 10 * 3 (number of digits - 1)
+    // sizeof(jlong) * 8 / 10 * 3 + 2 (number of digits + \0)
+    char deadline[sizeof(jlong) * 8 / 10 * 3 + 2];
```
Comment (a-siva) [inline shell/platform/darwin/ios/framework/Source/vsync_waiter_ios.mm]: ditto comment about these two lines.

File: shell/common/animator.cc [hunk 3]
```diff
@@ -41,6 +41,12 @@ void Animator::Start() {
   RequestFrame();
 }
 
+// This ID is used by the timeline component to correctly align
+// GPU Workloads events with their respective Framework Workload.
+const char* Animator::FrameId() {
```
Comment (rmacnak-google) [inline shell/common/animator.cc]: Why is this alternating instead of increasing?

File: shell/platform/android/vsync_waiter_android.cc (hunk 1 above)
Comment (B3rn475) [reply shell/platform/android/vsync_waiter_android.cc]: done

File: shell/platform/darwin/ios/framework/Source/vsync_waiter_ios.mm (hunk 2 above)
Comment (B3rn475) [reply shell/platform/darwin/ios/framework/Source/vsync_waiter_ios.mm]: done

File: shell/common/animator.cc (hunk 3 above)
Comment (B3rn475) [reply shell/common/animator.cc]: This values is used has a helper by the Frame alignment code to correctly pair `Framework Workload` events with the relative `GPU Workload`.
https://github.com/dart-lang/sdk/blob/master/runtime/observatory/web/timeline.js#L284

Due the fact that we can add just strings to the Timeline Events, alternating two constant values avoids us to print the value to it.

Comment (rmacnak-google) [reply shell/common/animator.cc]: Maybe FrameParity instead then, since this doesn't identify a unique frame.

Comment (B3rn475) [reply shell/common/animator.cc]: done

**Comment by B3rn475** [comment]
Rebasing

**Pull request merged**